### PR TITLE
[EGD-2489] Enable full newlib

### PR DESCRIPTION
**Pull request by alekrudnik**
Newlib nano is an optimized for size version of newlib libc library.
While newlib nano is smaller in size, it is also noticeably slower than
the full version - https://interrupt.memfault.com/blog/memcpy-newlib-nano

Newlib full is more consistent w c11 stanard, including
* full newlib provides native possiblity to printf 64-variables,
including double and float.
* calling destructors of global objectes ate system exit
* probaly many more...

There was hard fault while trying to enable full newlib whithout
recompilling memcpy implementation with -mno-unaligned-access compile
option.
It was failing for missalinged memory access during memcpy.
Similar issue is described here -
https://bugs.launchpad.net/gcc-arm-embedded/+bug/1857469

### CODE REVIEW

**Review by SP2FET — APPROVED** [review]

**Review by mpsm — APPROVED** [review]

**Review by lucckb — APPROVED** [review]

**Review by niepiekm — APPROVED** [review]

**Comment by alekrudnik** [comment]
Please do not merge until confirmed without approval !

File: board/rt1051/ldscripts/libs.ld
Comment (alekrudnik) [inline board/rt1051/ldscripts/libs.ld:13]: comment just to block auto merge

File: board/rt1051/memcpy-armv7m.S
Comment (niepiekm) [inline board/rt1051/memcpy-armv7m.S:2]: Instead of this assembly version, I'd like to suggest using a plain C version from here: 
https://github.com/travisg/newos/blob/master/lib/libc/string/memcpy.c
It's also used in [Android](https://android.googlesource.com/kernel/lk/+/master/lib/libc/string/memcpy.c).
It's short, readable and optimized for handling aligned and unaligned addresses.
And it compiles to [fewer instructions](https://gcc.godbolt.org/z/n4r1jdc46) than this one.

**Pull request merged**
